pilot/pkg/bootstrap/server.go

```diff
@@ -1424,6 +1442,41 @@ func (s *Server) initEventHandlers() error {
 	return nil
 }
 
+func (s *Server) initSidecarInjector(args *PilotArgs) error {
+	injectPath := args.InjectionOptions.InjectionDirectory
+	if injectPath == "" {
+		log.Infof("Skipping sidecar injector, injection path not specified")
+		return nil
+	}
+	// If the injection path exists, we will set up injection
+	if _, err := os.Stat(injectPath); !os.IsNotExist(err) {
+		dir, err := pilotDnsCertDir()
+		if err != nil {
+			return err
+		}
+		parameters := inject.WebhookParameters{
+			ConfigFile: filepath.Join(injectPath, "config"),
+			ValuesFile: filepath.Join(injectPath, "values"),
+			MeshFile:   args.Mesh.ConfigFile,
+			CertFile:   filepath.Join(dir, "cert-chain.pem"),
+			KeyFile:    filepath.Join(dir, "key.pem"),
+			Port:       args.InjectionOptions.Port,
+		}
+
+		wh, err := inject.NewWebhook(parameters)
+		if err != nil {
+			return fmt.Errorf("failed to create injection webhook: %v", err)
+		}
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go wh.Run(stop)
```

Comment:
Could not find the cert reconciling loop

Reply:
This is intended. See https://github.com/istio/istio/issues/17061 for more info. The webhook cert will be manged by istioctl/the operator. The cert has a 5 year TTL anyways so we don't really need such active management in pilot?

Reply:
Is the ttl configurable?

Reply:
I don't think so @lei-tang can you verify

Reply:
The DNS certificate signed by Kubernetes CA has a 5 year TTL and is not configurable.